src/rust/scheduler/handle.rs

```diff
@@ -55,26 +52,24 @@ pub struct YielderHandle {
 /// Associate Functions for Task Handlers
 impl TaskHandle {
     /// Creates a new Task Handle.
-    pub fn new(task_id: u64, index: usize, page: WakerPageRef) -> Self {
-        Self { task_id, index, page }
+    pub fn new(task_id: u64, waker_page_ref: WakerPageRef, waker_page_offset: usize) -> Self {
+        Self { task_id, waker_page_ref, waker_page_offset }
     }
 
     /// Queries whether or not the coroutine in the Task has completed.
     pub fn has_completed(&self) -> bool {
-        let subpage_ix: usize = self.index & (WAKER_BIT_LENGTH - 1);
-        self.page.has_completed(subpage_ix)
+        self.waker_page_ref.has_completed(self.waker_page_offset)
```

Comment:
Is this mask unnecessary?

Reply:
It is necessary becasue it allows us to compute waker page offset for the task from the pin slab index. With this PR, I just moved that existing computation to a common place.

Reply:
right so don't you need to mask waker_page_offset with WAKER_BIT_LENGTH - 1 still?

Reply:
This masking is still happening, but at a centralized place in `scheduler.rs` now where this function is called.